RELEASE_NOTES.txt

```diff
@@ -1,6 +1,138 @@
+* Fixed idle_verify hash conflicts for expiring entries.
+* Fixed in-memory server node crashing by TcpIgniteClient.putAllConflict() if cache objects transformation applied.
+* Fixed log message about sending and receiving partition file during snapshot restore.
+* Fixed log message of snapshot create operation with the warning result.
+* Fixed negative duration in the SQL query system view.
+* Fixed node crash due to SecurityContext not being found during discovery message processing.
+* Fixed opencensus-exporter-trace-zipkin dependency scope.
+* Fixed possibility to create two caches with the same schemas and index names.
+* Fixed possible failures of cache operations when performed through a thin client with Partition Awareness enabled.
+* Fixed result printing of the control.sh performance statistics subcommands.
+* Fixed server nodes crashing if the cache directory contains illegal chars.
+* Fixed snapshot creation failing when check throws an exception.
+* Fixed snapshot restore metrics is case of an error on prepare.
+* Fixed snapshot restore rollback if start cache stage failed.
+* Fixed startup on Java 21.
+* Fixed usage of sandbox to create CacheStore.
+* Fixed usage of vulnerable init setting in H2 connection.
+* Fixed writing post-handshake message in NIO server.
+* Removed deprecated "ssl.key.algorithm" Ignite system property - "ssl.KeyManagerFactory.algorithm" is used instead.
+* Removed the GridSslContextFactory from public API.
+* Updated Snappy-java dependency version to 1.1.10.4.
+* Updated Spring dependency version to 5.2.25.RELEASE.
+* Updated guava dependency version to 32.1.2.
+* Updated jetty dependency version to 9.4.53.v20231009.
+* Updated postgresql dependency version to 42.6.0.
+* Updated snappy dependency version to 1.1.10.4.
+* Updated zookeeper dependency version to 3.8.3.
+* Updated zstd dependency version to 1.5.5.
+
+SQL Calcite:
+* Added ENFORCE_JOIN_ORDER hint.
+* Added FORCE_INDEX/NO_INDEX hints.
+* Added events on query execution (EVT_CACHE_QUERY_OBJECT_READ, EVT_SQL_QUERY_EXECUTION).
+* Added explicit commons-codec dependency.
+* Added local flag support.
+* Added long-runing queries and big result-set queries warnings.
+* Added metrics for CalciteQueryExecutor thread pool.
+* Added partition pruning.
+* Added queries timeout support.
+* Added sensitive information hiding when IGNITE_TO_STRING_INCLUDE_SENSITIVE property is set to false.
+* Added support of setting partitions in SqlFieldsQuery.
+* Fixed CREATE TABLE ... WITH affinity_key option validation.
+* Fixed NPE on write query plan to performance statistics.
+* Fixed SQL metrics (added parsing metrics for the Calcite-based engine, fixed execution metrics).
+* Fixed failure on planning when cache size exceeds Integer.MAX_VALUE.
+* Fixed list of reserved keywords.
+* Fixed memory leak in MailboxRegistryImpl#remotes.
+* Fixed performance statistics gathering.
+* Fixed placeholder values from ModifyNode (backport of IGNITE-19096).
```

Comment:
Remove "backport of IGNITE-19096", it's a ticket from Ignite-3

Reply:
Done.